src/client/features/search/index.js

```diff
@@ -95,22 +96,42 @@ class Search extends React.Component {
     this.getSearchResult();
   }
 
+  setAndSubmitSearchQuery(searchTerm) {
+    const newQueryState = _.assign({}, this.state.query);
+    newQueryState.q = searchTerm;
+    this.setState({ query: newQueryState }, () => this.submitSearchQuery());
+  }
+
   render() {
     const props = this.props;
     const state = this.state;
 
+    const exampleSearches = ['TP53', 'Glycolysis', 'Ethanol'];
+
+    let examples = exampleSearches.map(search => {
+      const newQueryState = _.assign({}, state.query);
+      newQueryState.q = search;
+      return h('span.search-example', { onClick: () => this.setAndSubmitSearchQuery(search) }, search);
+    });
+
+    let i = 1;
+    while (i < examples.length) {
+      examples.splice(i, 0, ', ');
+      i += 2;
+    }
```

Comment:
what is this while loop supposed to do?

Reply:
The example searches underneath the search bar is produced from an array of suggested search queries in the render function. After the map happens, I want unclickable commas separating the clickable spans, which is what that while loop does. I'm sure there's a better way though.

The essential problem is taking an array and splicing a value between all entries,
`[1, 2, 3]` --> `[1, val, 2, val, 3]`

Reply:
(1) Make a function that returns the `h()` part for an example.

Basically you can create functions for simple React "components", e.g.

```js
let Foo = props => h('div.foo', { onBar: props.onBar });
```

(2) Add the examples manually, e.g.

```js
let examples = [
  h( Example, { q: 'p53' } ),
  h('span', ', '),
  h( Example, { q: 'pcna' } )
];
```

Reply:
Fixed.